### PR TITLE
Add last commit hash to error handler

### PR DESCRIPTION
**Pull request by fxha**
| Q                | A
| ---------------- | ---
| License          | MIT

### Description

Add the last commit hash to the error handler report and `--version` command if not build by us. I also added a release guide.

@Jocs Do we need to add `brew` instructions?

### CODE REVIEW

**Comment by fxha** [comment]
@Jocs Why `yarn` automatically updates the packages on AppVeyor/Travis? Can we disable it? 

E.g. Installed `electron-builder` version was `20.36.2` (according `package.json`) and CI failed because `yarn` installed version `20.38.0` and they break backward compatibility - renamed AppImage output file.

File: doc/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Steps to release Mark Text
+
+- Create a release candidate
+  - Create branch `release-v%version%`
+  - Set environment variable `MARKTEXT_IS_OFFICIAL_RELEASE` to `1` (default on AppVeyor and Travis CI)
+  - Ensure [changelog](https://github.com/marktext/marktext/blob/master/.github/CHANGELOG.md) is up-to-date
+  - Bump version
```
Comment (Jocs) [inline doc/RELEASE.md]: Bump the version in package.json and changelog?

File: doc/RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  - Update all `README.md` files
+  - Bump Flathub version ([marktext.appdata.xml](https://github.com/marktext/marktext/blob/master/resources/linux/marktext.appdata.xml))
+  - Create commit `release version %version%`
+  - Ensure all tests pass
+  - A new draft release should be available or create one
+- Publish GitHub release
+  - Add git tag `v%version%`
+  - Add changelog
+  - Add SHA256 checksums
```
Comment (Jocs) [inline doc/RELEASE.md]: Sorry, I still didn't know how and where will us use the `SHA256` checksums? 😄

**Comment by Jocs** [comment]
>Do we need to add brew instructions?

I will add instructions on how to update `marktext` in the readme file.

>Why yarn automatically updates the packages on AppVeyor/Travis? Can we disable it?

Some documents https://docs.npmjs.com/files/package-lock.json https://yarnpkg.com/lang/en/docs/yarn-lock/

I think we need to add a `yarn.lock` file.

File: doc/RELEASE.md (hunk 2 above)
Comment (fxha) [reply doc/RELEASE.md]: It is intended to verify the downloaded file(s). The checksum is calculated by AppVeyor or Travis CI; it should be calculated before/after uploading the binaries.

**Comment by fxha** [comment]
> I will add instructions on how to update marktext in the readme file.

I mean how to update the brew package or is it build from `master`?

**Comment by Jocs** [comment]
>I mean how to update the brew package or is it build from master?

we already has the [brew link](https://brew.sh/) in the readme file, i think user can look up this website.

**Pull request merged**